### PR TITLE
supporting method for submitting whole build(imagestream and buildlog)

### PR DESCRIPTION
**Pull request by harshad16**
Depends-On: https://github.com/thoth-station/user-api/pull/490

supporting method for submitting whole build(imagestream and buildlog)
- Would like the review on the support for build-watcher method
Signed-off-by: Harshad Reddy Nalla <hnalla@redhat.com>

Incomplete PR.
Required:
The swagger-codegen is to be rerun for update on thamos swagger_client with user-api new update.
https://github.com/thoth-station/user-api/pull/490

Related-to: 
- https://github.com/thoth-station/user-api/pull/490
- https://github.com/thoth-station/build-watcher/pull/44

### CODE REVIEW

File: thamos/lib.py [hunk 1]
```diff
@@ -417,6 +374,56 @@ def image_analysis(
...
+    environment_type: str,
+    registry_user: str = None,
+    registry_password: str = None,
+    verify_tls: bool = True,
+    nowait: bool = False,
+    force: bool = False,
+    debug: bool = False,
+) -> typing.Union[typing.Dict, str]:
+    """Submit a build for analysis to Thoth."""
+    if not base_image and not base_image:
```
Comment (fridex) [inline thamos/lib.py]: :eyes:

File: thamos/lib.py [hunk 2]
```diff
@@ -417,6 +374,56 @@ def image_analysis(
...
+        raise ValueError("No build info provided")
+
+    api_instance = BuildAnalysisApi(api_client)
+    if not image:
+        response = api_instance.post_build(
+            log_info=buildlog, base_image=base_image, debug=debug, verify_tls=verify_tls, force=force
+        )
+    if registry_user or registry_password:
```
Comment (fridex) [inline thamos/lib.py]: This branching does not look correct to me. There can happen 2 requests?

Comment (harshad16) [reply thamos/lib.py]: I will assume here that you are referencing to the outgoing 2 different API calls.

According to me, we need different API calls to be triggered suffice different scenarios.
The two different scenarios:
- The build is successful in the namespace we want BW to submit  **output Imagestream**, **base image stream**, and **Build log**.
- The build is unsuccessful in the namespace we want BW  to submit **base Imagestream**, and **build log**.

If this is not you were referring to. My apologies, can you rephrase the comment :) @fridex

Comment (fridex) [reply thamos/lib.py]: What about providing just 1 endpoint which will act based on the input?

* if all 3 (output imagestream, base image stream and build log) are provided
* if any 2 are provided
* if just one is provided

It will ensure that data are consistent and success/failure can be a flag. Might be easier to maintain the logic (maybe)?

BTW, as Christoph pointed out in user-api PR - it could worth to have output imagestream, base image stream, and build log part of the JSON input and have others parameters (from API design POV).

File: thamos/lib.py [hunk 3]
```diff
@@ -433,6 +440,51 @@ def image_analysis(
...
+    """Submit a build image and logs for analysis to Thoth."""
+    if not build_detail:
+        raise ValueError("No build info provided")
+
+    api_instance = BuildAnalysisApi(api_client)
+    if registry_user or registry_password:
+        # Swagger client handles None in a different way - we need to explicitly avoid passing
+        # registry user and registry password if they are not set.
+        response = api_instance.post_build(
+            body=build_detail,
```
Comment (fridex) [inline thamos/lib.py]: Can this be constructed explicitly? In other words - I, as a user of thamos.lib, I don't know what to pass to this function (`build_details` is too generic on this level). Can there be passed `log`, base image, output image explicitly as arguments and then, the request (which is a detail for thamos.lib consumer), is constructed in the python code of this function?

Comment (harshad16) [reply thamos/lib.py]: @fridex  Do you mean having build_log:object, output_image:string and base_image:string as the input of this function and inside the function logic we pass them as the build_detail?

Comment (fridex) [reply thamos/lib.py]: Yes - the `build_detail` would be constructed internally inside the function.

Comment (harshad16) [reply thamos/lib.py]: @fridex, the method is now updated

Comment (harshad16) [reply thamos/lib.py]: @fridex the method is now updated
